src/subsystem/can/commands.cpp

```diff
@@ -3,72 +3,371 @@
+int can_send(Node device, Command command, int num_bytes, unsigned int data) {
+    //Set status to unset
+    status = CAN_Status::UNSET;
+
 #ifdef ONBOARD_CAN_BUS
     //Socket is not open, nothing can send
     if (!socket_open) { return 1; }
     
     //Create the frame, based on specific cases for simplicity
-    canfd_frame frame;
-    switch (command) {
-        case Command::SET_AXIS_REQUESTED_STATE:
-            frame = get_can_frame(0, device, command, num_bytes, data);
-            break;
-        case Command::SET_INPUT_VEL:
-            frame = get_can_frame(0, device, command, num_bytes, data);
-            break;
-        case Command::SET_CONTROLLER_MODES:
-            frame = get_can_frame(4, device, command, num_bytes, data);
-            break;
-        default:
-            return 1;
+    can_frame frame;
+    if (command == Command::SET_CONTROLLER_MODES) {
+        frame = get_can_frame(4, device, command, num_bytes, data);
+    }
+    else {
+        frame = get_can_frame(0, device, command, num_bytes, data);
     }
 
     //Write the frame
     if (write(can_socket, &frame, 16) != 16) {
-        //Failed to write
+        status = CAN_Status::FAILED_WRITE;
         return 1;
     }
 #endif
     //Successfully sent
+    status = CAN_Status::SUCCESS;
     return 0;
 }
 
+//Send out a can request read message
+int can_request(Node device, Command command) {
+    //Set status to unset
+    status = CAN_Status::UNSET;
+
+#ifdef ONBOARD_CAN_BUS
+    //Socket is not open, nothing can send
+    if (!socket_open) { return 1; }
+    
+    //Create the frame
+    can_frame frame;
+    frame = get_can_request_frame(0, device, command);
+
+    //Write the frame
+    if (write(can_socket, &frame, 16) != 16) {
+        status = CAN_Status::FAILED_REQUEST;
+        return 1;
+    }
+#endif
+    //Successfully sent
+    status = CAN_Status::SUCCESS;
+    return 0;
+}
+
+//Request to read a CAN message, with a float output
+float can_read_float(Node device, Command command) {
+    //Read value
+    unsigned int received = 0;
+    received = can_receive(device, command);
+
+    //Convert convert from big endian (conversion is symmetrical) if not from an odrive
+    if (!is_odrive_device(device)) { received = get_big_endian(received); }
+    
+    //Convert unsigned int back to float, then return
+    union { unsigned int ul; float f; } conv = { .ul = received };
+    return conv.f;
+}
+
+//Request to read a CAN message, with an int output
+int can_read_int(Node device, Command command) {
+    //Read value
+    unsigned int received = 0;
+    received = can_receive(device, command);
+
+    //Convert convert from big endian (conversion is symmetrical) if not from an odrive, and return
+    if (!is_odrive_device(device)) { received = get_big_endian(received); }
+    return (int)received;
+}
+
+//Request to read a CAN message, with an long long output
+long long can_read_long(Node device, Command command) {
```

Comment:
In cases where the bit size of the integer matters a lot, you can use types like `int64_t` since the other types are all platform-dependent.

Reply:
Okay I will change that in the future. I also plan on making a single can_receive command that puts the information into a byte array, but for now this shouldn't be necessary.